R/scripts/build_US_setup.R

```diff
@@ -9,49 +42,96 @@ option_list = list(
+                                   variables="B01003_001", year=2018, 
+                                   keep_geo_vars=TRUE, geometry=FALSE, show_call=TRUE)
 census_data <- census_data %>%
-  filter(USPS %in% filterUSPS) %>%
-  select(USPS,GEOID,POP10) %>%
-  mutate(GEOID = substr(GEOID,1,5))
+  rename(population=estimate, geoid=GEOID) %>%
+  select(geoid, population) %>%
+  mutate(geoid = substr(geoid,1,5))
+
+# Add USPS column
+data(fips_codes)
+fips_geoid_codes <- mutate(fips_codes, geoid=paste0(state_code,county_code)) %>% 
+  group_by(geoid) %>% 
+  summarize(USPS=unique(state))
+census_data <- census_data %>% left_join(fips_geoid_codes, by="geoid")
 
+# Make each territory one county.
+# Puerto Rico is the only one in the 2018 ACS estimates right now. Aggregate it.
+# Keeping the other territories in the aggregation just in case they're there in the future.
 name_changer <- setNames(
-  unique(census_data$GEOID),
-  unique(census_data$GEOID)
+  unique(census_data$geoid),
+  unique(census_data$geoid)
 )
-name_changer[grepl("^60",name_changer)] <- "60000"
-name_changer[grepl("^66",name_changer)] <- "66000"
-name_changer[grepl("^69",name_changer)] <- "69000"
-name_changer[grepl("^72",name_changer)] <- "72000"
-name_changer[grepl("^78",name_changer)] <- "78000"
+name_changer[grepl("^60",name_changer)] <- "60000" # Amerian Samoa
+name_changer[grepl("^66",name_changer)] <- "66000" # Guam
+name_changer[grepl("^69",name_changer)] <- "69000" # Northern Mariana Islands
+name_changer[grepl("^72",name_changer)] <- "72000" # Puerto Rico
+name_changer[grepl("^78",name_changer)] <- "78000" # Virgin Islands
 
 census_data <- census_data %>%
-  mutate(GEOID = name_changer[GEOID]) %>%
-  group_by(GEOID) %>%
-  summarize(USPS = unique(USPS),POP10 = sum(POP10)) %>%
-  arrange(POP10)
+  mutate(geoid = name_changer[geoid]) %>%
+  group_by(geoid) %>%
+  summarize(USPS = unique(USPS), population = sum(population))
+
+# Territory populations (except Puerto Rico) taken from from https://www.census.gov/prod/cen2010/cph-2-1.pdf
+terr_census_data <- readr::read_csv(paste(opt$p,"sample_data","united-states-commutes","census_tracts_island_areas_2010.csv",sep='/'))
+
+census_data <- terr_census_data %>% 
+  filter(length(filterUSPS) == 0 | ((USPS %in% filterUSPS) & !(USPS %in% census_data)))%>%
+  rbind(census_data)
+
+# Sort by population
+census_data <- census_data %>%
+  arrange(population)
+
+if (!is.null(config$spatial_setup$popnodes)) {
+  names(census_data)[names(census_data) == "population"] <- config$spatial_setup$popnodes
+}
 
+
+if (length(config$spatial_setup$geodata) > 0) {
+  geodata_file <- config$spatial_setup$geodata
+} else {
+  geodata_file <- 'geodata.csv'
+}
+write.csv(file = file.path(outdir, geodata_file), census_data, row.names=FALSE)
+print(paste("Wrote geodata file:", file.path(outdir, geodata_file)))
+
+# COMMUTE DATA
+commute_data <- readr::read_csv(paste(opt$p,"sample_data","united-states-commutes","commute_data.csv",sep='/'))
```

Comment:
just checking: does opt$p need to be defined above?

Reply:
The default is "COVIDScenarioPipeline" if it's not provided by user on the command line. Should be line 39.